packages/loader/container-loader/src/container.ts

```diff
@@ -1560,7 +1560,7 @@ export class Container
 			// If we are trying to pause at a specific sequence number, ensure the latest snapshot is not newer than the desired sequence number.
 			if (loadMode.opsBeforeReturn === "sequenceNumber") {
 				assert(loadToSequenceNumber !== undefined, "sequenceNumber should be defined");
-				if (this.deltaManager.lastSequenceNumber > loadToSequenceNumber) {
+				if (dmAttributes.sequenceNumber > loadToSequenceNumber) {
```

Comment:
This is interesting and probably merits more investigation.  Probably the right thing here would be `attributes.sequenceNumber`, modify `dmAttributes.sequenceNumber` to be the specified value, and add handling below to block replay of ops beyond the requested sequence number.  Otherwise we might believe we can't load at the specified sequence number even when the snapshot is before it, just because we saved some ops that could be played on top of it.  This caching of ops is in a weird spot though so need to better understand what their expectations are.

Probably just log a backlog item for this though - I don't think we care too much about this particular scenario for migration purposes and if/when we fix it then we'll just be successful in more cases.  Probably do add a comment in this PR though explaining the relationship with savedOps.